src/main/java/io/github/thebusybiscuit/slimefun4/utils/tags/SlimefunTag.java

```diff
@@ -0,0 +1,312 @@
+     */
+    TERRACOTTA,
+
+    /**
+     * All ice variants, normal, packed, blue and whatever else there is.
+     */
+    ICE_VARIANTS,
+
+    /**
+     * All stone variants, normal, andesite, diorite, granite and whatever else may come.
+     */
+    STONE_VARIANTS,
+
+    /**
+     * All variants of concrete powder.
+     * Can you believe there is no tag for this already?
+     */
+    CONCRETE_POWDERS,
+
+    /**
+     * Materials which are sensitive to break.
+     * Things like Saplings or Pressure plates which break as well when you break
+     * the block beneath them.
+     */
+    SENSITIVE_MATERIALS,
+
+    /**
+     * These Materials are sensitive to fluids, they cannot be {@link Waterlogged}
+     * and would break in contact with water.
+     */
+    FLUID_SENSITIVE_MATERIALS,
+
+    /**
+     * These materials are just unbreakable, like bedrock for example.
+     */
+    UNBREAKABLE_MATERIALS,
+
+    /**
+     * Materials which cannot be reliably placed using a {@link BlockPlacer}.
+     */
+    BLOCK_PLACER_IGNORED_MATERIALS,
+
+    /**
+     * All materials which the {@link ExplosiveShovel} can break.
+     */
+    EXPLOSIVE_SHOVEL_BLOCKS,
+
+    /**
+     * All materials (ores) which the {@link PickaxeOfVeinMining} recognizes.
+     */
+    PICKAXE_OF_VEIN_MINING_BLOCKS,
+
+    /**
+     * All materials (ores) which the {@link PickaxeOfTheSeeker} recognizes.
+     */
+    PICKAXE_OF_THE_SEEKER_BLOCKS,
+
+    /**
+     * All materials which the {@link SmeltersPickaxe} will try and smelt.
+     */
+    SMELTERS_PICKAXE_BLOCKS,
+
+    /**
+     * All materials (ores) which the {@link IndustrialMiner} will try to mine.
+     */
+    INDUSTRIAL_MINER_ORES,
+
+    /**
+     * All materials (crops) which the {@link CropGrowthAccelerator} will recognize.
+     */
+    CROP_GROWTH_ACCELERATOR_BLOCKS,
+
+    /**
+     * All <strong>strong</strong> materials which the {@link ClimbingPick} is able to climb.
+     */
+    CLIMBING_PICK_STRONG_SURFACES,
+
+    /**
+     * All <strong>weak</strong> materials which the {@link ClimbingPick} is able to climb.
+     */
+    CLIMBING_PICK_WEAK_SURFACES,
+
+    /**
+     * This {@link SlimefunTag} holds all surfaces for the {@link ClimbingPick}.
+     * This is an aggregation of {@code CLIMBING_PICK_STRONG_SURFACES} and {@code CLIMBING_PICK_WEAK_SURFACES}
+     */
+    CLIMBING_PICK_SURFACES,
+
+    /**
+     * All materials (ores) which trigger the Talisman of the Caveman.
+     */
+    CAVEMAN_TALISMAN_TRIGGERS;
+
+    private static final Map<String, SlimefunTag> nameLookup = new HashMap<>();
+    public static final SlimefunTag[] valuesCache = values();
+
+    static {
+        for (SlimefunTag tag : valuesCache) {
+            nameLookup.put(tag.name(), tag);
+        }
+    }
+
+    private final NamespacedKey key;
+    private final Set<Material> includedMaterials = new HashSet<>();
+    private final Set<Tag<Material>> additionalTags = new HashSet<>();
```

Comment:
Isn't Tag an enum? So why don't you use EnumSet here?

Reply:
Tag is not an enum.